### PR TITLE
RIA-8083 RIA-8084 RIA-8085 - AiP Notification (Email/SMS)

### PR DESCRIPTION
**Pull request by javierhmcts**
### Jira link (if applicable)

https://tools.hmcts.net/jira/browse/RIA-8083

### Change description ###

Notif. implementation and testing

### Checklist

- [x] commit messages are meaningful and follow good commit message guidelines
- [ ] README and other documentation has been updated / added (if needed)
- [x] tests have been updated / new tests has been added (if needed)
- [ ] Does this PR introduce a breaking change

### CODE REVIEW

File: src/main/java/uk/gov/hmcts/reform/iacaseapi/domain/handlers/presubmit/RequestFeeRemissionAipPreparer.java [hunk 1]
```diff
@@ -107,9 +93,11 @@ public PreSubmitCallbackResponse<AsylumCase> handle(
                     asylumCase.write(PREVIOUS_REMISSION_DETAILS, remissionDetailsAppender.getRemissions());
                     assignLateRemissionValuesToRemissionValues(asylumCase);
                     clearPreviousAndLateRemissionFields(asylumCase);
+                    asylumCase.write(IS_LATE_REMISSION_REQUEST, YesOrNo.YES);
```
Comment (MiguelMartinRojas) [inline src/main/java/uk/gov/hmcts/reform/iacaseapi/domain/handlers/presubmit/RequestFeeRemissionAipPreparer.java:96]: just for clarity. can you move the write above the assignLateRemissionValuesToRemissionValues function?, same below.

Comment (ThomasKKC) [inline src/main/java/uk/gov/hmcts/reform/iacaseapi/domain/handlers/presubmit/RequestFeeRemissionAipPreparer.java:96]: What is the reason for this flag?

Comment (javierhmcts) [reply src/main/java/uk/gov/hmcts/reform/iacaseapi/domain/handlers/presubmit/RequestFeeRemissionAipPreparer.java:96]: Purpose of this flag as discussed with Ivan is to differentiate in the notification handler whether a record remission decision is coming from standard REMISSION or REFUND request (where refund means that is a late remission request AFTER appeal has been submitted) @ThomasKKC

**Review by ThomasKKC — APPROVED** [review]

**Review by MiguelMartinRojas — APPROVED** [review]
LGTM

**Review by MiguelMartinRojas — APPROVED** [review]

**Review by ThomasKKC — APPROVED** [review]

**Pull request merged**